src/app/workspace/organization/detail/payments-panel/payments-list/payments-list.component.ts

```diff
@@ -17,18 +17,85 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Component, OnInit } from '@angular/core';
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {OrganizationModel} from "../../../../../core/store/organizations/organization.model";
+import {Subscription} from "rxjs/Subscription";
+import {ActionsSubject, Store} from "@ngrx/store";
+import {Router} from "@angular/router";
+import {I18n} from "@ngx-translate/i18n-polyfill";
+import {AppState} from "../../../../../core/store/app.state";
+import {selectOrganizationByWorkspace} from "../../../../../core/store/organizations/organizations.state";
+import {isNullOrUndefined} from "util";
+import {filter} from "rxjs/operators";
+import {PaymentModel} from "../../../../../core/store/organizations/payment/payment.model";
+import {selectPaymentsByOrganizationIdSorted} from "../../../../../core/store/organizations/payment/payments.state";
+import {PaymentsAction, PaymentsActionType} from "../../../../../core/store/organizations/payment/payments.action";
+import {ServiceLimitsAction} from "../../../../../core/store/organizations/service-limits/service-limits.action";
 
 @Component({
   selector: 'payments-list',
   templateUrl: './payments-list.component.html',
   styleUrls: ['./payments-list.component.scss']
 })
-export class PaymentsListComponent implements OnInit {
+export class PaymentsListComponent implements OnInit, OnDestroy {
 
-  constructor() { }
+  private organization: OrganizationModel;
+  private organizationSubscription: Subscription;
+
+  public payments: PaymentModel[];
+  private paymentsSubscription: Subscription;
+
+  private getPaymentSubscription: Subscription;
```

Comment:
If you have multiple subscriptions in the component and the only place where you unsubscribe from them is `ngOnDestroy` method, you can combine them in a single subscription.

The field definition would look like this:

```typescript
private subscriptions = new Subscription();
```

Payment subscription (or any other subscription) would look like this:
```typescript
this.subscriptions.add(
  this.store.select(selectPaymentsByOrganizationIdSorted(this.organization.id))
      .pipe(filter(payments => !isNullOrUndefined(payments) && payments.length > 0))
      .subscribe(payments => this.payments = payments)
);
```

And `ngOnDestroy` method would look like this:
```typescript
public ngOnDestroy() {
  this.subscriptions.unsubscribe();
}
```

Reply:
Good hint for future!